server/chat-commands/core.ts

```diff
@@ -349,29 +348,30 @@ exports.commands = {
 	authlist: 'authority',
 	authority(target, room, user, connection) {
 		if (target) {
-			let targetRoom = Rooms.search(target);
-			let availableRoom = targetRoom && targetRoom.checkModjoin(user);
+			const targetRoom = Rooms.search(target);
+			const availableRoom = targetRoom?.checkModjoin(user);
 			if (targetRoom && availableRoom) return this.parse(`/roomauth1 ${target}`);
 			return this.parse(`/userauth ${target}`);
 		}
 		/** @type {{[k: string]: string[]}} */
-		let rankLists = {};
-		let ranks = Object.keys(Config.groups);
-		for (let u in Users.usergroups) {
-			let rank = Users.usergroups[u].charAt(0);
+		const rankLists: AnyObject = {};
+		const ranks = Object.keys(Config.groups);
+		for (const u in Users.usergroups) {
+			const rank = Users.usergroups[u].charAt(0);
 			if (rank === ' ' || rank === '+') continue;
 			// In case the usergroups.csv file is not proper, we check for the server ranks.
 			if (ranks.includes(rank)) {
-				let name = Users.usergroups[u].substr(1);
-				if (!rankLists[rank]) rankLists[rank] = [];
+				const name: string = Users.usergroups[u].substr(1);
```

Comment:
```suggestion
				const name = Users.usergroups[u].substr(1);
```

TS has type inference for this.